fennel/featuresets/test_invalid_featureset.py

```diff
@@ -238,3 +245,114 @@ class UserInfo:
         str(e.value)
         == "Feature `extractors` in `UserInfo` has a reserved name `extractors`."
     )
+
+
+@featureset
+class UserInfo2:
+    user_id: int = feature(id=1)
+    home_geoid: int = feature(id=2)
+    age: int = feature(id=3).extract(field=UserInfoDataset.age, default=0)  # type: ignore
+    credit_score: int = feature(id=4)
+
+
+@mock
+def test_tp(client):
+    client.sync(datasets=[UserInfoDataset], featuresets=[UserInfo2])
+
+
+@mock
+def test_invalid_autogenerated_extractors(client):
+    with pytest.raises(TypeError) as e:
+
+        @featureset
+        class UserInfo:
+            user_id: int = feature(id=1)
+            home_geoid: int = feature(id=2)
+            age: int = feature(id=3).extract(field=UserInfoDataset.age)
+            credit_score: int = feature(id=4)
+
+    # age should be optional[int]
+    if sys.version_info < (3, 9):
+        assert (
+            str(e.value)
+            == "Feature `UserInfo.age` has type `<class 'int'>` but expectected type `typing.Union[int, NoneType]`"
+        )
+    else:
+        assert (
+            str(e.value)
+            == "Feature `UserInfo.age` has type `<class 'int'>` but expectected type `typing.Optional[int]`"
```

Comment:
I'm a bit confused on the point of this PR. In this example the dataset field is non optional and so intuitively to me the extracted feature would also be non optional. For values where the lookup doesn't return anything, this is what the default is for. So if you'll always have an int from the ds or a default int, what is this automatic relaxing to an optional type giving us / why are we doing it? Is it for the case when there is no default value (if so, why is this generalized to even when a default is present?)

Reply:
I dont get your question, this example does not have a default value, and hence age should be Optional[int], if you specify a default it should be in